packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -90,6 +90,26 @@ const getMediaGalleryEntries = (product, optionCodes, optionSelections) => {
     return value;
 };
 
+const getBreadcrumbCategoryId = categories => {
+    const breadcrumbSet = new Set();
+    categories.forEach(({ breadcrumbs }) => {
+        // breadcrumbs can be `null`...
+        (breadcrumbs || []).forEach(({ category_id }) =>
+            breadcrumbSet.add(category_id)
+        );
+    });
+
+    // Filter out category data for categories that exist as breadcrumbs.
+    const leafCategories = categories.filter(
+        category => !breadcrumbSet.has(category.id)
+    );
+
+    // Return the first category id of the potential leaf categories.
+    // This can potentially be jarring if the user navigated to a product and
+    // the breadcrumbs don't represent that navigation path.
```

Comment:
I don't think it'll be jarring at all. Breadcrumbs should be a canonical path to the product as understood by the server (or in this case, estimated by the client using a simple heuristic: first category wins). We don't need to be in the business of showing the user their navigation history. That's what browser history and browser chrome are for.

At any rate, I like this a lot more than the dropdown.

Reply:
Well I certainly don't want to lead opinion through this comment -- I can remove it and simply say "Until we can get the single canonical breadcrumb path to a product we will just select the first leaf category"